### PR TITLE
Add posix_utils.osReleaseFile

### PR DESCRIPTION
**Pull request by juancarlospaco**
- Add `posix_utils.osReleaseFile`. Reliably identify the details of your Linux.
- https://www.freedesktop.org/software/systemd/man/os-release.html#Example
- Kinda based but not copied from https://docs.python.org/3.10/library/platform.html#platform.freedesktop_os_release
- Changelog, since, docs with links, etc.
- Works on Alpine and bigger distros.

### CODE REVIEW

File: lib/posix/posix_utils.nim [hunk 1]
```diff
@@ -107,3 +109,17 @@ proc mkdtemp*(prefix: string): string =
   if mkdtemp(tmpl) == nil:
     raise newException(OSError, $strerror(errno))
   return $tmpl
+
+proc osReleaseFile*(): Config {.since: (1, 5).} =
+  ## Get system identification from `os-release` file and return it as a `Config`.
```
Comment (ringabout) [inline lib/posix/posix_utils.nim]:
```suggestion
  ## Gets system identification from `os-release` file and returns it as a `Config`.
```

File: lib/posix/posix_utils.nim [hunk 2]
```diff
@@ -107,3 +109,17 @@ proc mkdtemp*(prefix: string): string =
   if mkdtemp(tmpl) == nil:
     raise newException(OSError, $strerror(errno))
   return $tmpl
+
+proc osReleaseFile*(): Config {.since: (1, 5).} =
+  ## Gets system identification from `os-release` file and return it as a `Config`.
```
Comment (ringabout) [inline lib/posix/posix_utils.nim]:
```suggestion
  ## Gets system identification from `os-release` file and returns it as a `Config`.
```

**Comment by xflywind** [comment]
Maybe you could make a test?

File: lib/posix/posix_utils.nim [hunk 3]
```diff
@@ -107,3 +109,23 @@ proc mkdtemp*(prefix: string): string =
...
+  ## Gets system identification from `os-release` file and returns it as a `Config`.
+  ## The `os-release` file is an official Freedesktop.org open standard.
+  ## Available in Linux distributions, except Android and Android-based.
+  ## `os-release` file is not available on Windows and OS X by design.
+  ## * https://www.freedesktop.org/software/systemd/man/os-release.html
+  runnableExamples:
+    import parsecfg
+    when defined(linux):
+      let data = osReleaseFile()
+      echo data.getSectionValue("","NAME")  ## the data is up to each distro.
```
Comment (ringabout) [inline lib/posix/posix_utils.nim]: please use assert, at least you could use `assert sth.len > 0`

**Comment by Araq** [comment]
If this is Linux specific code, it should go to ``linux.nim``. Posix != Linux.

**Comment by juancarlospaco** [comment]
@Araq BSD too, I added it explicitly to the doc.

File: lib/posix/posix_utils.nim [hunk 4]
```diff
@@ -11,7 +11,9 @@
 
 # Where possible, contribute OS-independent procs in `os <os.html>`_ instead.
 
-import posix
+import posix, parsecfg, os
+import std/private/since
+export Config
```
Comment (Araq) [inline lib/posix/posix_utils.nim]:
```suggestion
```

File: lib/posix/posix_utils.nim [hunk 5]
```diff
@@ -107,3 +109,21 @@ proc mkdtemp*(prefix: string): string =
   if mkdtemp(tmpl) == nil:
     raise newException(OSError, $strerror(errno))
   return $tmpl
+
+proc osReleaseFile*(): Config {.since: (1, 5).} =
+  ## Gets system identification from `os-release` file and returns it as a `Config`.
```
Comment (Araq) [inline lib/posix/posix_utils.nim]:
```suggestion
  ## Gets system identification from `os-release` file and returns it as a `parsecfg.Config`.
  ## You also need to import the `parsecfg` module to gain access to this object.
```

File: changelog.md [hunk 6]
```diff
@@ -85,6 +85,11 @@
 - Added `mimetypes.mimesMaxLen` thats equal to the length of the longest "mime" from `mimes`.
 
 
+
+- Added `posix_utils.osReleaseFile` to get system identification from `os-release` file on Linux.
```
Comment (Araq) [inline changelog.md]:
```suggestion
- Added `posix_utils.osReleaseFile` to get system identification from `os-release` file on Linux and the BSDs.
```

**Pull request merged**